source/Calamari/Kubernetes/SetupKubectlAuthentication.cs

```diff
@@ -18,18 +18,21 @@ namespace Calamari.Kubernetes
 {
     public class SetupKubectlAuthentication
     {
+        public const string KubernetesTentacleTargetTypeId = "KubernetesTentacle";
+        public const string SkippingAuthenticationMessage = "Running inside Kubernetes cluster, using pod service account for authentication";
+
         readonly IVariables variables;
         readonly ILog log;
         readonly ICommandLineRunner commandLineRunner;
-        private readonly Kubectl kubectl;
+        readonly IKubectl kubectl;
         readonly Dictionary<string, string> environmentVars;
         readonly string workingDirectory;
         string aws;
 
         public SetupKubectlAuthentication(IVariables variables,
```

Comment:
This is becoming a mammoth file. Not a blocker but we may want to break it up a little if possible (or not, just boy-scout rule )

Reply:
That's fair enough, I'll take a look 👍

Reply:
I'm going to do this in a separate few PRs. It's a bit of a scary part of the code because it does a lot of crazy stuff and while there are higher level auth integration tests, there are no lower level more detailed unit tests.

I've got a PR to add a bunch of Unit Tests and a subsequent PR to split out all the different auth context providers to make this class a lot neater.

Reply:
>I've got a PR to add a bunch of Unit Tests

Nice, this was what I was hoping that breaking it up would lead to. The more we can make use of unit tests over integration tests the more stable coverage we will have.